app/deploy_test.go

```diff
@@ -869,3 +863,28 @@ func (s *S) TestRebuild(c *check.C) {
 	c.Assert(writer.String(), check.Equals, "Rebuild deploy called")
 	c.Assert(imgID, check.Equals, "app-image")
 }
+
+func (s *S) TestRollbackUpdate(c *check.C) {
```

Comment:
I think you could add more tests to this func (the same cases your did for the handler, but calling app.RollBackUpdate directly) as @cezarsa. The rest lgtm